src/ansys_sphinx_theme/theme/ansys_sphinx_theme/static/js/fuse_search.js

```diff
@@ -0,0 +1,131 @@
+    resultsContainer.innerHTML = "";
+
+    if (results.length === 0) {
+      displayNoResultsMessage(resultsContainer);
+      return;
+    }
+
+    // Show the results container if there's a query
+    if (query === "") {
+      resultsContainer.style.display = "none";
+      return;
+    }
+    resultsContainer.style.display = "block";
+
+    // Populate results
+    results.forEach((result) => {
+      const { title, text, href } = result.item;
+      const item = createResultItem(title, text, href, query);
+      resultsContainer.appendChild(item);
+    });
+  }
+
+  // Display "No matched documents" message
+  function displayNoResultsMessage(container) {
+    const noResultsMessage = document.createElement("div");
+    noResultsMessage.className = "no-results";
+    noResultsMessage.textContent = "No matched documents";
+    container.appendChild(noResultsMessage);
+  }
+
+  // Create and return a result item
+  function createResultItem(title, text, href, query) {
+    const item = document.createElement("div");
+    item.className = "result-item";
+
+    const highlightedTitle = highlightTerms(title, query);
+    const highlightedText = highlightTerms(text, query);
+
+    item.innerHTML = `
+      <div class="result-title">${highlightedTitle}</div>
+      <div class="result-text">${highlightedText}</div>
+    `;
+    item.setAttribute("data-href", href);
+
+    // Navigate to the result's href on click
+    item.addEventListener("click", () => navigateToHref(href));
+    return item;
+  }
+
+  // Highlight matching terms in search results
+  function highlightTerms(text, query) {
+    if (!query.trim()) return text;
+    const words = query.trim().split(/\s+/);
+    const escapedWords = words.map((word) =>
+      word.replace(/[-\/\\^$*+?.()|[\]{}]/g, "\\$&"),
+    );
+    const regex = new RegExp(`(${escapedWords.join("|")})`, "gi");
+    return text.replace(regex, '<span class="highlight">$1</span>');
+  }
+
+  // Navigate to the href
+  function navigateToHref(href) {
+    const baseUrl = window.location.origin;
+    const relativeUrl = href.startsWith("/") ? href : `/${href}`;
+    window.location.href = new URL(relativeUrl, baseUrl).href;
+  }
+
+  // Event listeners
+  const searchBox = document
+    .getElementById("search-bar")
+    .querySelector(".bd-search input");
+
+  // Handle input in the search box
+  searchBox.addEventListener("input", function () {
+    const query = this.value.trim();
+    if (query.length < parseInt(min_chars_for_search)) {
+      document.getElementById("results").innerHTML = "";
+      return;
+    }
+    performSearch(query);
+  });
+
+  // Handle Enter key press in the search box
+  searchBox.addEventListener("keydown", function (event) {
+    if (event.key === "Enter") {
+      const firstResult = document.querySelector(".result-item");
+      if (firstResult) {
+        navigateToHref(firstResult.getAttribute("data-href"));
+      }
+      event.preventDefault();
+    }
+  });
```

Comment:
Could we handle `ArrowDown` and `ArrowUp` events to highlight a certain result before pressing enter? Similarly to the current behavior with Meilisearch.

Reply:
Also, using a `switch-case` layout here can improve the readability.

Reply:
sure i will add it.

Reply:
Added in [906a397](https://github.com/ansys/ansys-sphinx-theme/pull/496/commits/906a39798f6725d85a559a441d81ed25f016db3d)